roles/servicetelemetry/tasks/component_prometheus.yml

```diff
@@ -7,90 +7,133 @@
-          - use
-
-  - name: Bind the local prometheus SA to our new role
-    k8s:
-      definition:
-        apiVersion: rbac.authorization.k8s.io/v1
-        kind: RoleBinding
-        metadata:
-          name: prometheus-k8s-stf
-          namespace: '{{ ansible_operator_meta.namespace }}'
-        roleRef:
-          apiGroup: rbac.authorization.k8s.io
-          kind: Role
-          name: prometheus-stf
-        subjects:
-        - kind: ServiceAccount
-          name: prometheus-k8s
-          namespace: '{{ ansible_operator_meta.namespace }}'
-  when:
-    - observability_strategy in ['use_redhat', 'use_hybrid']
-
-- name: Bind the local prometheus SA to prometheus cluster role (for oauth perms)
+- name: Create ClusterRole/prometheus-stf for non-resource URL /metrics access
+  k8s:
+    definition:
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: ClusterRole
+      metadata:
+        name: prometheus-stf
+      rules:
+      - apiGroups:
+        - ""
+        resources:
+        - nodes/metrics
+        verbs:
+        - get
+      - nonResourceURLs:
+        - /metrics
+        verbs:
+        - get
+      - apiGroups:
+        - authentication.k8s.io
+        resources:
+        - tokenreviews
+        verbs:
+        - create
+      - apiGroups:
+        - authorization.k8s.io
+        resources:
+        - subjectaccessreviews
+        verbs:
+        - create
+      - apiGroups:
+        - ""
+        resources:
+        - namespaces
+        verbs:
+        - get
+      - apiGroups:
+        - security.openshift.io
+        resourceNames:
+        - nonroot
+        - nonroot-v2
+        resources:
+        - securitycontextconstraints
+        verbs:
+        - use
+
+- name: Create ClusterRoleBinding/prometheus-stf
   k8s:
     definition:
       apiVersion: rbac.authorization.k8s.io/v1
       kind: ClusterRoleBinding
       metadata:
-        name: prometheus-k8s-{{ ansible_operator_meta.namespace }}
-        namespace: '{{ ansible_operator_meta.namespace }}'
+        name: prometheus-stf
       roleRef:
         apiGroup: rbac.authorization.k8s.io
         kind: ClusterRole
-        name: prometheus-k8s
+        name: prometheus-stf
       subjects:
       - kind: ServiceAccount
-        name: prometheus-k8s
+        name: prometheus-stf
+        namespace: '{{ ansible_operator_meta.namespace }}'
+
+- name: Create Role/prometheus-stf for Prometheus operations
+  k8s:
+    definition:
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: Role
+      metadata:
+        name: prometheus-stf
+        namespace: '{{ ansible_operator_meta.namespace }}'
+      rules:
+      - apiGroups:
+        - ""
+        resources:
+        - services
+        - endpoints
+        - pods
+        verbs:
+        - get
+        - list
+        - watch
+      - apiGroups:
+        - extensions
+        - networking.k8s.io
+        resources:
+        - ingresses
+        verbs:
+        - get
+        - list
+        - watch
```

Comment:
The `prometheus-k8s` role doesn't include these, so not sure if we're using this because we permit UI access via oauth-proxy perhaps?

CC: @csibbitt

Reply:
Other prometheus-k8s objects are using this same Role configuration across the OCP cluster. I'm going to leave it for now until we can do further investigation about what CRDs etc are using this interface. If we're truly not using it and it won't cause issues, then I'll be more comfortable removing it at that point.